core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -327,8 +328,9 @@ private boolean isInliningCandidate(Expression expression, ProjectNode node)
             verify(AstUtils.preOrder(expression).noneMatch(TryExpression.class::isInstance));
 
             // candidate symbols for inlining are
-            //   1. references to simple constants or symbol references
```

Comment:
Good call. I understand these two rules have similarities in terms of qualifying inlining candidates, but I would prefer to handle that rule in a separate PR since this one is dealing with predicate pushdown issue